semgrep-core/Analyzing/Dataflow_constness.ml

```diff
@@ -140,10 +140,10 @@ let literal_of_int i =
   let i_str = string_of_int i in
   (* TODO: use proper token when possible? *)
```

Comment:
Change this comment? It looks like comparing large integers now relies on having the token be the string representation of the integer. It might also be better to call it "i_str" instead of tok. I was a little confused about what we were going to do if the integers/floats were too large

Reply:
no the comment is about something else, that is when building AST elements, we should try to reuse tokens from
the existing AST to get better error reporting.
If the integer/floats are too large, we default to string comparisons.

Reply:
Actually we would have to be more careful now, if we evaluate `2 * 42` we cannot just reuse the token of either `2` or  `42`, we would need to change it's `str` field, since this may be used for matching.